src/main/java/org/codeforamerica/messaging/providers/twilio/TwilioCallbackController.java

```diff
@@ -37,33 +38,25 @@ public ResponseEntity<Object> updateStatus(HttpServletRequest request) {
             return new ResponseEntity<>("Unauthorized", HttpStatus.UNAUTHORIZED);
         }
         String rawMessageStatus = request.getParameter("MessageStatus");
-        if (ignorable(rawMessageStatus)) {
-            return ResponseEntity.ok().build();
-        }
-        SmsMessage smsMessage = smsMessageRepository.findFirstByProviderMessageId(providerMessageId);
-        MessageStatus newSmsStatus = mapTwilioStatusToMessageStatus(rawMessageStatus);
-        MessageStatus currentSmsStatus = smsMessage.getMessage().getSmsStatus();
-        if (newSmsStatus.isAfter(currentSmsStatus)) {
-            log.info("Updating status. Provider message id: {}, new status: {}", providerMessageId, newSmsStatus);
-            smsMessage.getMessage().setRawSmsStatus(rawMessageStatus);
-            smsMessage.getMessage().setSmsStatus(newSmsStatus);
-            smsMessage.setFromPhone(request.getParameter("From"));
-            if (hadError(smsMessage)) {
-                smsMessage.setProviderError(buildProviderError(request));
-            }
-            smsMessageRepository.save(smsMessage);
-        } else {
-            log.info("Ignoring earlier status {}", newSmsStatus);
+        if (!ignorable(rawMessageStatus)) {
```

Comment:
I would _prefer_ if `ignorable` could be renamed to make the logic read more simply, e.g. `requiresUpdate` instead of "not ignorable", but it isn't important.